adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,301 @@
+package resetdigital
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"text/template"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v2/adapters"
+	"github.com/prebid/prebid-server/v2/config"
+	"github.com/prebid/prebid-server/v2/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint    *template.Template
+	endpointUri string
+}
+
+type resetDigitalRequest struct {
+	Site resetDigitalSite  `json:"site"`
+	Imps []resetDigitalImp `json:"imps"`
+}
+type resetDigitalSite struct {
+	Domain   string `json:"domain"`
+	Referrer string `json:"referrer"`
+}
+type resetDigitalImp struct {
+	ZoneID     resetDigitalImpZone    `json:"zone_id"`
+	BidID      string                 `json:"bid_id"`
+	ImpID      string                 `json:"imp_id"`
+	Ext        resetDigitalImpExt     `json:"ext"`
+	MediaTypes resetDigitalMediaTypes `json:"media_types"`
+}
+type resetDigitalImpZone struct {
+	PlacementID string `json:"placementId"`
+}
+type resetDigitalImpExt struct {
+	Gpid string `json:"gpid"`
+}
+type resetDigitalMediaTypes struct {
+	Banner resetDigitalMediaType `json:"banner"`
+	Video  resetDigitalMediaType `json:"video"`
+}
+type resetDigitalMediaType struct {
+	Sizes [][]int64 `json:"sizes"`
+}
+
+type resetDigitalBidResponse struct {
+	Bids []resetDigitalBid `json:"bids"`
+}
+type resetDigitalBid struct {
+	BidID string  `json:"bid_id"`
+	ImpID string  `json:"imp_id"`
+	CPM   float64 `json:"cpm"`
+	CID   string  `json:"cid,omitempty"`
+	CrID  string  `json:"crid,omitempty"`
+	AdID  string  `json:"adid"`
+	W     string  `json:"w,omitempty"`
+	H     string  `json:"h,omitempty"`
+	Seat  string  `json:"seat"`
+	HTML  string  `json:"html"`
+}
+
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter, server config.Server) (adapters.Bidder, error) {
+	template, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+	bidder := &adapter{
+		endpoint: template,
+	}
+	return bidder, nil
+}
+
+func getHeaders(request *openrtb2.BidRequest) http.Header {
+	headers := http.Header{}
+
+	if request != nil && request.Device != nil && request.Site != nil { // what about request.App? Do we need to do something different with Referrer in the app case assuming we care about app?
+		addNonEmptyHeaders(&headers, map[string]string{
+			"Referer":         request.Site.Page,
+			"Accept-Language": request.Device.Language,
+			"User-Agent":      request.Device.UA,
+			"X-Forwarded-For": request.Device.IP,
+			"X-Real-Ip":       request.Device.IP,
+			"Content-Type":    "application/json;charset=utf-8",
+			"Accept":          "application/json",
+		})
+	}
```

Comment:
This conditional only adds the headers if both site and device are not nil.
Is it possible that one of those is present but the other is nil?
If so, you should probably have two separate calls to `addNonEmptyHeaders` where one is wrapped in a `if request != nil && request.Device != nil` conditional and the other in a `if request != nil && request.Site != nil` conditional.